### PR TITLE
VarDict module utils

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY

Refactor `VarDict` out of the `ModuleHelper` umbrella, making it simpler and easier to integrate without any dependencies to MH itself.

##### ISSUE TYPE

- New Module/Plugin Pull Request

##### COMPONENT NAME

plugins/module_utils/vardict.py

### CODE REVIEW

File: plugins/module_utils/vardict.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        return self.change and (self.initial_value != self.value)
+
+    @property
+    def diff_result(self):
+        if self.diff and self.has_changed:
+            return {'before': self.initial_value, 'after': self.value}
+        return
+
+    def __str__(self):
+        return "<_Variable: value={0}, initial={1}, diff={2}, output={3}, change={4}, verbosity={5}>".format(
```
Comment (felixfontein) [inline plugins/module_utils/vardict.py:80]:
```suggestion
        return "<_Variable: value={0!r}, initial={1}, diff={2}, output={3}, change={4}, verbosity={5}>".format(
```

File: plugins/module_utils/vardict.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    def setverbosity(self, v):
+        if not (0 <= v <= 4):
+            raise ValueError("verbosity must be an int in the range 0 to 4")
+        self._verbosity = v
+
+    change = property(getchange, setchange)
+    verbosity = property(getverbosity, setverbosity)
+
+    def set(self, output=None, diff=None, change=None, fact=None, initial_value=NOTHING, verbosity=None):
```
Comment (felixfontein) [inline plugins/module_utils/vardict.py:45]: Some documentation on what the types should be would be helpful.

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution!

File: plugins/module_utils/vardict.py (hunk 2 above)
Comment (russoz) [reply plugins/module_utils/vardict.py]: Done, although users should not be calling any method directly into an instance of `_Variable`, they should be interacting only with the `VarDict` class.

Comment (felixfontein) [reply plugins/module_utils/vardict.py]: True, but if you look at the `VarDict` function that passes everything to this function, you only see `**kwargs` being passed without any details. So if someone looks for documentation, they end up looking here.

**Review by felixfontein — APPROVED** [review]
Will merge for 7.1.0.

**Comment by felixfontein** [comment]
@russoz thanks for your contribution!

**Pull request merged**